internal/operands/tekton-tasks/reconcile.go

```diff
@@ -307,3 +276,21 @@ func getTektonTasksNamespace(request *common.Request) string {
 	}
 	return request.Instance.Namespace
 }
+
+func filterResources[PT interface {
+	*T
+	metav1.Object
+}, T any](items []T, trimSuffix bool) []T {
+	var results []T
+	for i := range items {
+		object := PT(&items[i])
```

Comment:
Isn't this an unchecked cast by the way?

Reply:
I've used @lyarwood's piece of code that is simpler.

Reply:
Just to reply. This is not an unchecked cast. The yype system guarantees that it will always be correct.
`PT` is a `*T` , `items[i]` has type `T`, and `&items[i]` has type `*T`.